types/redis/redis-tests.ts

```diff
@@ -171,3 +172,6 @@ client.zincrby('a', 1, 'b', strCallback);
 client.ZINCRBY('a', 1, 'b', strCallback);
 
 client.flushdb(okCallback);
+
+// hgetall callback's result can be null
+client.hgetall(str, nullableCallback);
```

Comment:
because:

```ts
type Callback = (x: number) => void;

const foo: Callback = (x: number | null): void => {}; // This is fine
```

`foo` is assignable to callback because it accepts a number _and more_. so it meets the requirements.

just like `(reply: T | null) => void` is assignable to `(reply: T) => void` here